### DISCUSSION TITLE
Dynamic trigger + Job DI

### DISCUSSION THREAD

**Discussion by strigefleur** [quartznet/quartznet#1471] [Q&A]
Hello.
I'm trying to utilize following scenario: create a trigger with different params at runtime (by user request) and schedule it to the same job.
Works fine if I use options way to create a job but it prevents me from using injected services.
So I'm trying to switch to `UseMicrosoftDependencyInjectionJobFactory`, and now I'm confused by the docs.

1. How do I attach my dynamically created trigger to the DI-instantiated job? Previously I used `jobKey` to get the job instance.
2. Should I add some `services.AddTransient<MyJob>` while having `q.UseMicrosoftDependencyInjectionJobFactory()` or not?

Thanks!

**Comment by jaumemilian** [comment]
Did you find the proper way to solve this use case?
I'm facing a similar scenario and the documentation is not really helping me on that

Thanks,
Jaume

**Comment by lahma** [comment]
You can register your jobs statically in DI config (probably inside `AddQuartz`, make sure jobs are durable so they do not need a trigger. Then you just inject `ISchedulerFactory` to place where you are going to add the triggers (say Controller instance) and `IScheduler scheduler = await factory.GetScheduler();` then you have `IScheduler` interface which you can use to add triggers dynamically, just need to know the job key.

DI config is for convenience (static setup), but if you need all the power `IScheduler` has the API services.
